packages/pangea-sdk/src/main/java/cloud/pangeacyber/pangea/intel/UserIntelClient.java

```diff
@@ -60,4 +62,25 @@ public UserPasswordBreachedResponse breached(UserPasswordBreachedRequest request
 		throws PangeaException, PangeaAPIException {
 		return doPost("/v1/password/breached", request, UserPasswordBreachedResponse.class);
 	}
+
+	public static PasswordStatus isPasswordBreached(UserPasswordBreachedResponse response, String hash)
+		throws PangeaException {
+		Map<String, Object> rawData = response.getResult().getRawData();
+		if (rawData == null) {
+			throw new PangeaException("Need raw data to check if hash is breached. Send request with raw=true", null);
+		}
+
+		Object hashData = rawData.get(hash);
+		if (hashData != null) {
+			// If hash is present in raw data, it's because it was breached
+			return PasswordStatus.BREACHED;
+		} else if (rawData.size() >= 1000) {
```

Comment:
if youre going to do this then you should be comparing each of the returned hashes with the original to see if it matches

Reply:
If I'm not wrong, raw result has a dictionary where full hash is the key and value is an object which has "count" information. So, what I did is to check if my full hash is present on that dictionary, if it's not present, I check the number of keys to decide.